app/src/main/java/com/example/firstapp/RegisterActivity.java

```diff
@@ -0,0 +1,159 @@
+                    }
+
+                }
+                Toast.makeText(RegisterActivity.this, "You must fill out all the fields", Toast.LENGTH_SHORT).show();
+
+            }
+        });
+
+        hideSoftKeyboard();
+
+    }
+
+    private void registerNewEmail(String email, String password) {
+        showDialog();
+
+        FirebaseAuth.getInstance().createUserWithEmailAndPassword(email, password).addOnCompleteListener(
+                new OnCompleteListener<AuthResult>() {
+                    @Override
+                    public void onComplete(@NonNull Task<AuthResult> task) {
+                        Log.d(TAG, "oncomplete: onComplete" + task.isComplete());
+
+                        if (task.isSuccessful()){
+                            Log.d(TAG, "oncomplete: AuthState" + FirebaseAuth.getInstance().getCurrentUser()
+                                    .getUid());
+
+                            FirebaseAuth.getInstance().signOut();
+                        }
+                        else {
+                            Toast.makeText(RegisterActivity.this, "Unable to Register",
+                                    Toast.LENGTH_SHORT).show();
+                        }
+                        hideDialog();
+                    }
+                }
+        );
+    }
+
+
+    /**
+     * Returns True if the user's email contains '@andela.com'
+     * @param email
+     * @return
+     */
+    private boolean isValidDomain(String email){
+        Log.d(TAG, "isValidDomain: verifying email has correct domain: " + email);
+        String domain = email.substring(email.indexOf("@") + 1).toLowerCase();
+        Log.d(TAG, "isValidDomain: users domain: " + domain);
+        return domain.equals(DOMAIN_NAME);
+    }
+
+    /**
+     * Redirects the user to the login screen
+     */
+    private void redirectLoginScreen(){
+        Log.d(TAG, "redirectLoginScreen: redirecting to login screen.");
+
+        Intent intent = new Intent(RegisterActivity.this, LoginActivity.class);
+        startActivity(intent);
+        finish();
+    }
+    /**
+     * Return true if @param 's1' matches @param 's2'
+     * @param s1
+     * @param s2
+     * @return
+     */
+    private boolean doStringsMatch(String s1, String s2){
+        return s1.equals(s2);
+    }
+
+    /**
+     * Return true if the @param is null
+     * @param string
+     * @return
+     */
+    private boolean isEmpty(String string){
+        return string.equals("");
+    }
+
+
+    private void showDialog(){
+        mProgressBar.setVisibility(View.VISIBLE);
+
+    }
+
+    private void hideDialog(){
+        if(mProgressBar.getVisibility() == View.VISIBLE){
+            mProgressBar.setVisibility(View.INVISIBLE);
+        }
+    }
+
+    private void hideSoftKeyboard(){
+        this.getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_HIDDEN);
+    }
+
```

Comment:
What do you thing about implementing databinding to make the code a lot cleaner? Otherwise LGTM